pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3628,6 +3628,8 @@ private PersistentReplicator getReplicatorReference(String replName, PersistentT
 
     private CompletableFuture<Void> updatePartitionedTopic(TopicName topicName, int numPartitions) {
         final String path = ZkAdminPaths.partitionedTopicPath(topicName);
+        PartitionedTopicMetadata metadata = getPartitionedTopicMetadata(topicName, false, false);
```

Comment:
I think it's better to get partition metadata asynchronously here in case deadlock might happen.